src/test/java/org/sagebionetworks/bridge/services/AdherenceServiceTest.java

```diff
@@ -323,29 +345,102 @@ public void updateAdherenceRecords_finishExistingSessionRecord() {
         assertEquals(captured.getInstanceGuid(), "sessionInstanceGuid");
         assertEquals(captured.getStartedOn(), STARTED_ON);
         assertEquals(captured.getFinishedOn(), FINISHED_ON);
-        assertTrue(captured.isDeclined()); // used the persisted record 
+        assertFalse(captured.isDeclined()); // we now calculate this on the submitted records 
         assertEquals(captured.getEventTimestamp(), EVENT_TS);
     }
     
+    @Test
+    public void updateAdherenceRecords_declineExistingSessionRecord() {
+        AdherenceRecordList list = mockRecordUpdate(
+                ar(STARTED_ON, FINISHED_ON, "AAA", true), 
+                ar(STARTED_ON, FINISHED_ON, "BBB", true), 
+                sar(null, null, false));
+        
+        MetadataContainer container = new MetadataContainer(mockScheduleService, list.getRecords());
+        service.updateSessionState(TEST_APP_ID, container, list.getRecords().get(0));
+        service.updateSessionState(TEST_APP_ID, container, list.getRecords().get(1));
+        
+        AdherenceRecord captured = Iterables.getFirst(container.getSessionUpdates(), null);
+        assertEquals(captured.getAppId(), TEST_APP_ID);
+        assertEquals(captured.getUserId(), TEST_USER_ID);
+        assertEquals(captured.getStudyId(), TEST_STUDY_ID);
+        assertEquals(captured.getInstanceGuid(), "sessionInstanceGuid");
+        assertEquals(captured.getStartedOn(), STARTED_ON);
+        assertEquals(captured.getFinishedOn(), FINISHED_ON);
+        assertTrue(captured.isDeclined()); // we now calculate this on the submitted records 
+        assertEquals(captured.getEventTimestamp(), EVENT_TS);
+    }
```

Comment:
remove me, this is redundant with the next test.

Reply:
changed it to demonstrate that some assessments marked as declined do not mark the session as declined.